samples: bluetooth: nrf_dm: Add energy consumption information. (#9310)
nrfconnect/sdk-nrf: samples/bluetooth/nrf_dm/README.rst

When a device does not respond within a certain period of time, it is removed from the list.
Also the last measurement results with the peers are stored in this list.

Energy consumption and the number of measurements performed in a given time depends on:

   * Advertiser settings.
   * Scanner settings.

>    * Extend of measurement window by ``extra_window_time_us`` in the ``dm_request`` structure.

Author: This is additional time that extends the measurement window. That's why I changed it to: `Additional size of measurement window by ``extra_window_time_us`` in the :c:struct:`dm_request` structure.`

>    * Additional size of measurement window by ``extra_window_time_us`` in the :c:struct:`dm_request` structure.

Reviewer: How about "Extended size" instead of "Additional size"?

Author: Sounds good.

The shorter the interval, the higher the frequency of broadcasting advertising packages.
This leads to a higher probability of synchronization and to a higher number of measurements taken at a given time.

> However, the higher number of packets transmitted also results in higher power consumption.

Reviewer 2: maybe we can add a note here saying that this decreases the latency until a new device is picked up and that.
Additionally, maybe we can link the relevant KConfigs here.

Author: Added additional note.

The scan interval and scan window parameters determine how often and how long the scanning device will listen for packets.
Like the short advertising interval, these values also have an impact on energy consumption.

Extending the measurement window will increase the probability of measurement success for coarse synchronization.
However, it affects the radio's turn-on time and thus, power consumption.

> 

Reviewer 2: Maybe we can also mention DM_MIN_TIME_BETWEEN_TIMESLOTS_US and DM_TIMESLOT_QUEUE_COUNT_SAME_PEER with which you can control how often the same peer is ranged.

When a device does not respond within a certain period of time, it is removed from the list.
Also the last measurement results with the peers are stored in this list.

> Energy consumption and the number of measurements performed in a given time depend on the following factors:

Reviewer 2: Maybe make this a separate section?

> Energy consumption and the number of measurements performed in a given time depend on:
> 
>    * Advertiser settings.
>    * Scanner settings.
>    * Extended size of measurement window by ``extra_window_time_us`` in the :c:struct:`dm_request` structure.

Reviewer: ```suggestion
Energy consumption and the number of measurements performed in a given time depend on the following factors:

* Advertiser settings.
* Scanner settings.
* Extended size of measurement window by ``extra_window_time_us`` in the :c:struct:`dm_request` structure.
```

> The shorter the interval, the higher the frequency of broadcasting advertising packages.

Reviewer 2: ```suggestion
The shorter the advertising interval, the higher the frequency at which the broadcaster sends out advertising packages.
```

> This leads to a higher probability of synchronization and to a higher number of measurements taken at a given time.

Reviewer 2: ```suggestion
This leads to a higher probability of synchronization and to a higher number of measurements taken at a given time and decreases the latency for the first measurement when the device is turned on or comes in reach.
```

However, the higher number of packets transmitted also results in higher power consumption.

The scan interval and scan window parameters determine how often and how long the scanning device will listen for packets.

> Like the short advertising interval, these values also have an impact on energy consumption.

Reviewer 2: ```suggestion
Like the short advertising interval, these values also have an impact on power consumption and influence how fast and how often devices are ranged.
```
I think technically energy cannot be consumed.

Author: Fixed.

> How often the measurement will be executed is also managed by :kconfig:option:`CONFIG_DM_MIN_TIME_BETWEEN_TIMESLOTS_US`.

Reviewer: ```suggestion
How often the measurement will be executed is also managed by the :kconfig:option:`CONFIG_DM_MIN_TIME_BETWEEN_TIMESLOTS_US` Kconfig option.
```

> How often the measurement will be executed is also managed by the :kconfig:option:`CONFIG_DM_MIN_TIME_BETWEEN_TIMESLOTS_US` Kconfig option.

Author: If `CONFIG_DM_TIMESLOT_RESCHEDULE` is disabled then the next timeslot will not be allocated by the dm module. The next measurement will take after the next synchronization. So how often the device is measured depends on the frequency of synchronization. 
`CONFIG_DM_TIMESLOT_QUEUE_COUNT_SAME_PEER` affects this indirectly.

> How often the measurement will be executed is also managed by the :kconfig:option:`CONFIG_DM_MIN_TIME_BETWEEN_TIMESLOTS_US` Kconfig option.

Reviewer 2: Should we also add a note about that config?

> How often the measurement will be executed is also managed by the :kconfig:option:`CONFIG_DM_MIN_TIME_BETWEEN_TIMESLOTS_US` Kconfig option.

Author: Good point. Added.

> The :kconfig:option:`CONFIG_DM_TIMESLOT_QUEUE_COUNT_SAME_PEER` and :kconfig:option:`CONFIG_DM_TIMESLOT_QUEUE_LENGTH` Kconfig options also indirectly affect this.

Reviewer: ```suggestion
The Kconfig options :kconfig:option:`CONFIG_DM_TIMESLOT_QUEUE_COUNT_SAME_PEER` and :kconfig:option:`CONFIG_DM_TIMESLOT_QUEUE_LENGTH` also have an indirect impact on this.
```

Extending the measurement window will increase the probability of measurement success for coarse synchronization.

> However, it affects the radio's turn-on time and thus, power consumption.

Reviewer: ```suggestion
However, it affects the turn-on time of the radio and thus, power consumption.
```

> At the time of application runtime, the additional delay of measurement execution is configured using ``start_delay_us`` in the :c:struct:`dm_request` structure.

Reviewer: ```suggestion
At the application runtime, the additional delay of measurement execution is configured using the ``start_delay_us`` parameter of the :c:struct:`dm_request` structure.
```

> This option helps adjust the synchronization.
> :kconfig:option:`CONFIG_DM_INITIATOR_DELAY_US` Kconfig options defines the initiator delay.
> Increasing these parameters increases the latency.

Reviewer: ```suggestion
This option helps in adjusting the synchronization.
The :kconfig:option:`CONFIG_DM_INITIATOR_DELAY_US` Kconfig option defines the initiator delay.
Increasing these parameters increases the latency.
```
Are the parameters increased or their values?

Author: It is about the value of these parameters.

When a device does not respond within a certain period of time, it is removed from the list.
Also the last measurement results with the peers are stored in this list.

Power consumption
*****************

Power consumption and the number of measurements performed in a given time depend on the following factors:

* Advertiser settings.
* Scanner settings.
* Extended size of measurement window by ``extra_window_time_us`` in the :c:struct:`dm_request` structure.

The shorter the advertising interval, the higher the frequency at which the broadcaster sends out advertising packages.

> This leads to a higher probability of synchronization and to a higher number of measurements taken at a given time and decreases the latency for the first measurement when the device is turned on or comes in reach.

Reviewer: ```suggestion
This leads to a higher probability of synchronization and to a higher number of measurements taken at a given time.
It also decreases the latency for the first measurement when the device is turned on or comes in reach.
```

However, the higher number of packets transmitted also results in higher power consumption.

The scan interval and scan window parameters determine how often and how long the scanning device will listen for packets.

> Like the short advertising interval, these values also have an impact on power consumption and influence how often devices are ranged.

Reviewer: ```suggestion
Like the short advertising interval, these values also have an impact on power consumption and the ranging frequency of devices.
```

> After the first measurement, the :kconfig:option:`CONFIG_DM_MIN_TIME_BETWEEN_TIMESLOTS_US` and the :kconfig:option:`CONFIG_DM_TIMESLOT_QUEUE_COUNT_SAME_PEER` Kconfig options define how fast the same device is measured again.

Author: Right, changed to _often_.

The Kconfig options :kconfig:option:`CONFIG_DM_TIMESLOT_QUEUE_COUNT_SAME_PEER` and :kconfig:option:`CONFIG_DM_TIMESLOT_QUEUE_LENGTH` define how many measurements are being scheduled.

> The more measurements that are being scheduled, the more measurements take place, which increases power consumption but gives more frequent rangings.

Reviewer: ```suggestion
The more measurements are scheduled, the more of them take place, which increases power consumption but gives more frequent rangings.
```

If the :kconfig:option:`CONFIG_DM_TIMESLOT_RESCHEDULE` Kconfig option is disabled, the next timeslot for measurement execution will be allocated after the measurement request is added using the :c:func:`dm_request_add` function.

> Extending the measurement window will increase the probability of measurement success for coarse synchronization.

Reviewer: ```suggestion
Extending the measurement window increases the probability of measurement success for coarse synchronization.
```
